ananas/src/main/java/iamutkarshtiwari/github/io/ananas/editimage/fragment/RotateFragment.java

```diff
@@ -119,79 +114,100 @@ public void onClick(View v) {
     public void onClick(View v) {
         int id = v.getId();
-        if (id == R.id.rotate_left) {
-            int updatedAngle = rotatePanel.getRotateAngle() - RIGHT_ANGLE;
-            rotatePanel.rotateImage(updatedAngle);
-        } else if (id == R.id.rotate_right) {
-            int updatedAngle = rotatePanel.getRotateAngle() + RIGHT_ANGLE;
-            rotatePanel.rotateImage(updatedAngle);
+        EditImageActivity activity;
+        if ((activity = getActivityInstance())!= null) {
+            if (id == R.id.rotate_left) {
+                int updatedAngle = activity.rotatePanel.getRotateAngle() - RIGHT_ANGLE;
+                activity.rotatePanel.rotateImage(updatedAngle);
+            } else if (id == R.id.rotate_right) {
+                int updatedAngle = activity.rotatePanel.getRotateAngle() + RIGHT_ANGLE;
+                activity.rotatePanel.rotateImage(updatedAngle);
+            }
         }
     }
 
     public void applyRotateImage() {
-        if (rotatePanel.getRotateAngle() == 0 || (rotatePanel.getRotateAngle() % 360) == 0) {
-            backToMain();
-        } else {
-            compositeDisposable.clear();
-            Disposable applyRotationDisposable = applyRotation(activity.getMainBit())
-                    .subscribeOn(Schedulers.computation())
-                    .observeOn(AndroidSchedulers.mainThread())
-                    .doOnSubscribe(subscriber -> loadingDialog.show())
-                    .doFinally(() -> loadingDialog.dismiss())
-                    .subscribe(processedBitmap -> {
-                        if (processedBitmap == null)
-                            return;
-
-                        applyAndExit(processedBitmap);
-                    }, e -> {
-                        // Do nothing on error
-                    });
-
-            compositeDisposable.add(applyRotationDisposable);
+        EditImageActivity activity;
+        if ((activity = getActivityInstance()) != null) {
+            if (activity.rotatePanel.getRotateAngle() == 0
+                    || (activity.rotatePanel.getRotateAngle() % 360) == 0) {
+                backToMain();
+            } else {
+                compositeDisposable.clear();
+                Disposable applyRotationDisposable = applyRotation(activity.getMainBit())
+                        .subscribeOn(Schedulers.computation())
+                        .observeOn(AndroidSchedulers.mainThread())
+                        .doOnSubscribe(subscriber -> {
+                            EditImageActivity activityInstance;
+                            if ((activityInstance = getActivityInstance()) != null)
+                                activityInstance.showLoadingDialog();
+                        })
+                        .doFinally(() -> {
+                            EditImageActivity activityInstance;
+                            if ((activityInstance = getActivityInstance()) != null)
+                                activityInstance.showLoadingDialog();
```

Comment:
is this change deliberate?
before this change the "doFinally" was running dismiss on the dialog, not show

Reply:
It should be dismiss, I'll fix and update the same